accepted/0003-plugin-manager.md

```diff
@@ -0,0 +1,153 @@
+- Start Date: 2019-12-30
+- RFC PR: #4
+- Mantis Issue: N/A
+
+# Summary
+Create a plugin manager that is capable of configuring the loading of plugins as
+well as providing an easier means of installing plugins. Furthermore, the plugin
+manager should be responsible for updating said plugins. This should be achieved
+by hosting a platform for developers to submit their plugins that can be
+verfied, signed, and easily maintained. More about managing and the necessary
+api in policy.
+
+# Motivation
+Varying from plugin to plugin, there are a variety of different intallation
+methods for each plugin. On top of that, once installed, there is no easy way to
+disable, or even configure, each individual plugin. In addition, if one plugin
+doesn't behave correctly because of another, it can become difficult to diagnose
+which plugins are causing the error. By implementing a plugin manager, it allows
+users to seamlessly install plugins and have finer control over the loading of
+their plugins. Furthermore, having a plugin manager opens up many more
+possibilties for both users and developers.
+
+More importantly, however, is the need for better discoverability and
+compatibilty. As OBS undergoes changes and updates, it could easily break
+different plugins. The plugin manager would be responsible for allowing users to
+better find the plugins they need as well as ensuring that they work with their
+version of OBS. To further reinforce compatability, their should also be
+different levels, such as stable, edge, and beta, again, more discussed in policy.
+
+# Design
+## Functionality
+### Loading
+Currently, when plugins cause issues, the only way to disable them is by
+outright uninstalling them. However, this could be mitigated at startup by
+having an internal record of which plugins are not to be run. By keeping a list -
+perhaps in the configuration file - of the disabled plugins, they could be
+ignored during initilizaiton.
+
+Furthermore, if a plugin causes a major failure while loading or doing use, it
```

Comment:
Addin in addition to this a performance measurement of which plugins cause the highest hits to system performance would be beneficial. This would help with diagnosing issues that are being caused by bad plugins not properly managing their resource utilization. Also knowing which plugins take the longest to load on OBS launch may become handy as well.

Reply:
OBS module load time is already logged in the OBS logs at the end of each OBS session under "obs_load_all_modules" in the profiler.